### DISCUSSION TITLE
Supabase migrations don't include table/column comments

### DISCUSSION THREAD

**Discussion by dawnmist** [supabase/supabase#19478] [Feature Requests]
# Bug report

## Describe the bug

Comments added to tables or columns are not included in the diff when a migration is created from that database. This results in being unable to document tables, as any comments in the dev db will be lost when it is next reset, and none of the dev db comments can be propagated to staging/prod because they're not part of the migration files.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Use the example countries SQL to create a new countries table with comment descriptions for the table and each column in your local dev db. (Accessible from a hosted supabase project => SQL editor => Welcome => Quickstart => Countries)
2. Run `supabase db diff -f add_countries` to generate a migration for that countries table.

## Expected behavior

Expect the generated migration to include the commands to create the table, any indexes that were defined, and all comments, as per:

```sql
CREATE TYPE "public"."continents" AS enum (
    'Africa',
    'Antarctica',
    'Asia',
    'Europe',
    'Oceania',
    'North America',
    'South America'
);

CREATE TABLE "public"."countries" (
    "id" bigint generated by DEFAULT AS identity NOT NULL,
    "name" text,
    "iso2" text NOT NULL,
    "iso3" text,
    "local_name" text,
    "continent" continents
);


CREATE UNIQUE INDEX countries_pkey ON public.countries USING btree (id);

ALTER TABLE "public"."countries"
ADD CONSTRAINT "countries_pkey" PRIMARY KEY USING INDEX "countries_pkey";

-- Everything below this line is missing from the migration, but ought to be present.
COMMENT ON TABLE countries IS 'Full list of countries.';
COMMENT ON COLUMN countries.name IS 'Full country name.';
COMMENT ON COLUMN countries.iso2 IS 'ISO 3166-1 alpha-2 code.';
COMMENT ON COLUMN countries.iso3 IS 'ISO 3166-1 alpha-3 code.';
COMMENT ON COLUMN countries.local_name IS 'Local variation of the name.';
```

## Actual behaviour

None of the 5 `COMMENT ON *` lines quoted above exist in the migration file created when using `supabase db diff`.

## System information

- OS: Debian 11
- Chrome 110.0.5481.177
- Version of supabase cli: 1.40.1
- Version of supabase-js: 2.10.0
- Version of Node.js: v18.12.1

## Additional context

I've tried using all 3 of `supabase db diff`, `supabase db diff --use-migra` and `supabase db diff --use-pgadmin`, but none of the generated migration files contain the expected comment lines.

**Comment by nrayburn-tech** [comment]
I don't think the underlying tooling currently supports comments as part of the migration, https://github.com/djrobstep/migra/issues/69.

**Comment by sweatybridge** [comment]
Yup, this is a known limitation with the diff tool we use.

Another difficulty in supporting this out of the box is that only [object owners](https://www.postgresql.org/docs/current/sql-comment.html) can set comments. Since the cli only runs migrations as the `postgres` user, setting comments won't work for objects owned by a custom role. Hence we decided not to include comments by default.

If comments are necessary, one workaround is to add them manually to your migration files. Note that these comments have to appear before any alter owner statements for it to work.

**Reply by myaroshefsky** [reply]
As it becomes more important for AI to understand our schemas, I wanted to +1 this feature request. Comments provide helpful context for our AI partners to understand and work with our database tables.
